File: uima-toolkit/src/main/scala/jenshaase/uimaScala/toolkit/type/Token.scala
/* First created by JCasGen Tue Apr 12 17:01:07 CEST 2011*/
package jenshaase.uimaScala.toolkit.type

import org.apache.uima.jcas.{JCas, JCasRegistry}
import org.apache.uima.jcas.cas.TOP_Type

import org.apache.uima.jcas.tcas.Annotation


/* Updated by JCasGen Tue Apr 12 17:01:07 CEST 2011
 * Xml Source ./uima-toolkit/src/main/resources/desc/type/toolkit-typ-system.xml
 * @generated */
class Token extends Annotation {

  /* @generated */
  def getTypeIndexID(): Integer = {
    Token.typeIndexID
  }

  /* Never called.  Disable default constructor
   * @generated */
  protected def this() = {}

  /* Initernal - constructor used by generator
   * @generated */
  def this(addr: Integer, type: TOP_TYPE) = {
    this(addr, type)
    readObject()
  }

  /* @generated */
  def this(jcas: JCas) = {
    this(jcas)
    readObject()
  }

  /* @generated */
  def this(jcas: JCas, begin: Integer, end: Integer) = {
    this(jcas)
    setBegin(begin)
    setEnd(end)
    readObject()
  }

  /* <!-- begin-user-doc -->
   * Write your own initialization here
   * <!-- end-user-doc -->
   * @generated modifiable */
  private def readObject(): Unit = {}


}

/* Updated by JCasGen Tue Apr 12 17:01:07 CEST 2011
 * Xml Source ./uima-toolkit/src/main/resources/desc/type/toolkit-typ-system.xml
 * @generated */
object Token  {

  /* A simple token annotation
   * @generated
   * @ordered */
  val typeIndexID = JCasRegistry.register(classOf[Token])

  /* @generated
   * @ordered */
  val type = typeIndexID


}

File: uima-toolkit/src/main/java/com/github/jenshaase/uimascala/toolkit/types/Token.java


/* First created by JCasGen Sun Jan 22 13:52:21 CET 2012 */
package com.github.jenshaase.uimascala.toolkit.types;

import org.apache.uima.jcas.JCas; 
import org.apache.uima.jcas.JCasRegistry;
import org.apache.uima.jcas.cas.TOP_Type;

import org.apache.uima.jcas.tcas.Annotation;


/** 
 * Updated by JCasGen Sun Jan 22 13:52:21 CET 2012
 * XML source: /home/jens/programming/scala/uimaScala/uima-toolkit/src/main/resources/desc/types/com.github.jenshaase.uimascala.toolkit.description.BasicTypeDescription.xml
 * @generated */
public class Token extends Annotation {
  /** @generated
   * @ordered 
   */
  public final static int typeIndexID = JCasRegistry.register(Token.class);
  /** @generated
   * @ordered 
   */
  public final static int type = typeIndexID;
  /** @generated  */
  public              int getTypeIndexID() {return typeIndexID;}
 
  /** Never called.  Disable default constructor
   * @generated */
  protected Token() {}
    
  /** Internal - constructor used by generator 
   * @generated */
  public Token(int addr, TOP_Type type) {
    super(addr, type);
    readObject();
  }
  
  /** @generated */
  public Token(JCas jcas) {
    super(jcas);
    readObject();   
  } 

  /** @generated */  
  public Token(JCas jcas, int begin, int end) {
    super(jcas);
    setBegin(begin);
    setEnd(end);
    readObject();
  }   

  /** <!-- begin-user-doc -->
    * Write your own initialization here
    * <!-- end-user-doc -->
  @generated modifiable */
  private void readObject() {}
     
 
    
  //*--------------*
  //* Feature: POS

  /** getter for POS - gets 
   * @generated */
  public String getPOS() {
    if (Token_Type.featOkTst && ((Token_Type)jcasType).casFeat_POS == null)
      jcasType.jcas.throwFeatMissing("POS", "com.github.jenshaase.uimascala.toolkit.types.Token");
    return jcasType.ll_cas.ll_getStringValue(addr, ((Token_Type)jcasType).casFeatCode_POS);}
    
  /** setter for POS - sets  
   * @generated */
  public void setPOS(String v) {
    if (Token_Type.featOkTst && ((Token_Type)jcasType).casFeat_POS == null)
      jcasType.jcas.throwFeatMissing("POS", "com.github.jenshaase.uimascala.toolkit.types.Token");
    jcasType.ll_cas.ll_setStringValue(addr, ((Token_Type)jcasType).casFeatCode_POS, v);}    
   
    
  //*--------------*
  //* Feature: lemma

  /** getter for lemma - gets 
   * @generated */
  public String getLemma() {
    if (Token_Type.featOkTst && ((Token_Type)jcasType).casFeat_lemma == null)
      jcasType.jcas.throwFeatMissing("lemma", "com.github.jenshaase.uimascala.toolkit.types.Token");
    return jcasType.ll_cas.ll_getStringValue(addr, ((Token_Type)jcasType).casFeatCode_lemma);}
    
  /** setter for lemma - sets  
   * @generated */
  public void setLemma(String v) {
    if (Token_Type.featOkTst && ((Token_Type)jcasType).casFeat_lemma == null)
      jcasType.jcas.throwFeatMissing("lemma", "com.github.jenshaase.uimascala.toolkit.types.Token");
    jcasType.ll_cas.ll_setStringValue(addr, ((Token_Type)jcasType).casFeatCode_lemma, v);}    
   
    
  //*--------------*
  //* Feature: stem

  /** getter for stem - gets 
   * @generated */
  public String getStem() {
    if (Token_Type.featOkTst && ((Token_Type)jcasType).casFeat_stem == null)
      jcasType.jcas.throwFeatMissing("stem", "com.github.jenshaase.uimascala.toolkit.types.Token");
    return jcasType.ll_cas.ll_getStringValue(addr, ((Token_Type)jcasType).casFeatCode_stem);}
    
  /** setter for stem - sets  
   * @generated */
  public void setStem(String v) {
    if (Token_Type.featOkTst && ((Token_Type)jcasType).casFeat_stem == null)
      jcasType.jcas.throwFeatMissing("stem", "com.github.jenshaase.uimascala.toolkit.types.Token");
    jcasType.ll_cas.ll_setStringValue(addr, ((Token_Type)jcasType).casFeatCode_stem, v);}    
  }

    